pyhive/hive.py

```diff
@@ -66,7 +66,7 @@ def connect(*args, **kwargs):
 class Connection(object):
     """Wraps a Thrift session"""
 
-    def __init__(self, host, port=10000, username=None, database='default', auth='NONE',
+    def __init__(self, host, kerberos_service_name, port=10000, username=None, password=None, database='default', auth='NONE',
```

Comment:
The `kerberos_service_name` param, what if there is not one provided? Not all users will be running kerberos. Also, new api additions should go at the end of the method, and maybe they should have `...=None` default values?